src/NUnitFramework/framework/Constraints/MsgUtils.cs

```diff
@@ -29,9 +29,24 @@
 namespace NUnit.Framework.Constraints
 {
     /// <summary>
+    /// Custom value formatter function
+    /// </summary>
+    /// <param name="val">The value</param>
+    /// <returns></returns>
+    public delegate string ValueFormatter(object val);
+
+    /// <summary>
+    /// Custom value formatter factory function
+    /// </summary>
+    /// <param name="next">The next formatter function</param>
+    /// <returns>ValueFormatter</returns>
+    /// <remarks>If the given formatter is unable to handle a certain format, it must call the next formatter in the chain</remarks>
+    public delegate ValueFormatter ValueFormatterFactory(ValueFormatter next);
+
+    /// <summary>
     /// Static methods used in creating messages
     /// </summary>
-    internal static class MsgUtils
+    public static class MsgUtils
```

Comment:
It's not the end of the world to expose this entire class publicly, but I'm not in love with it.